src/common/options.cc

```diff
@@ -7463,6 +7463,36 @@ static std::vector<Option> get_rbd_options() {
     .set_default(0)
     .set_description("the desired burst limit of write bytes"),
 
+    Option("rbd_qos_iops_burst_length", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
```

Comment:
Nit: instead of a `length` suffix, let's use `seconds` to be specific about the units.

Reply:
The burst length is not equal to the burst duration. 
when limit = 4000, burst=5000, length=10,
burst_seconds = (5000*10)/(5000-4000) = 50 s.

Reply:
What's the end-goal of the "length"? To me, it's confusing to have to think of this as a multiplier on the maximum burst rate. Is the goal of this really to be the rate maximum credit in the bucket? i.e. a bucket that can only hold 50,000 credits can sustain a burst of 5,000 IOPS over the limit of 4,000 IOPS for 50 seconds? If that's the case, how about `rbd_qos_iops_credit_limit` (and similar)?

Reply:
This design is inspired by libvirt and Huawei EVS.
https://libvirt.org/formatdomain.html
please search "total_bytes_sec_max_length"

https://support.huaweicloud.com/en-us/productdesc-evs/en-us_topic_0014580744.html
please search "Burst Capability and Principles"

Reply:
The libvirt domain XML specifies that the "total_XYZ_max_length" fields are in seconds. So I say again, either this field should represent the max number of seconds that a burst can occur for or, perhaps ideally, just the maximum number of credits the bucket can hold. The Huawei documents are similar (it talks about number of seconds at max burst limit and also the number of tokens/credits the bucket can hold).

Reply:
In fact, the design of QEMU and Huawei is the same
bucket capacity = max * max_ length （qemu）
bucket cpacity = burst * burst_ duration (huawei)
actually maximum duration in seconds = capacity / (burst - limit) 

set burst duration to 10 seconds，actually burst duration may be 50 seconds or  other values（decide by limit、burst、burst_duration），seems a little confuseds.

Reply:
We can use burst duration in seconds to describe the capacity control parameter just as them do，but users may be a little confused.

Reply:
I agree that `seconds` as a unit in confusing since it's only true when at maximum burst rate -- that's why I am saying it might be clearer that this is the bucket capacity.

Reply:
Maybe we can let capacity = burst  +（burst-limit）* (burst_seconds -1) .

Reply:
[root@wwc-test-x86-ceph-1 ~]# rbd config image set vms/test rbd_qos_iops_burst 1000
[root@wwc-test-x86-ceph-1 ~]# rbd config image set vms/test rbd_qos_iops_limit 500
[root@wwc-test-x86-ceph-1 ~]# rbd config image set vms/test rbd_qos_iops_burst_seconds 10
[root@wwc-test-x86-ceph-1 ~]# 
[root@wwc-test-x86-ceph-1 ~]# 
[root@wwc-test-x86-ceph-1 ~]# rbd bench --io-type write --io-size 4K --io-total 1G vms/test
bench  type write io_size 4096 io_threads 16 bytes 1073741824 pattern sequential
  SEC       OPS   OPS/SEC   BYTES/SEC
    1      1056   1074.23  4400047.70
    2      2048   1033.11  4231627.15
    3      3040   1019.42  4175561.90
    4      4048   1016.59  4163936.74
    5      5040   1011.68  4143850.06
    6      6048    998.48  4089759.89
    7      7040    998.48  4089759.90
    8      8048   1001.68  4102868.10
    9      9024    995.28  4076651.68
   10     10032    998.48  4089759.85
   11     10528    896.07  3670297.27
   12     11024    796.86  3263942.90
   13     11520    694.45  2844480.33
   14     12032    601.53  2463849.61
   15     12528    499.14  2044470.96
   16     13024    499.14  2044470.96
   17     13536    502.34  2057576.55
   18     14032    502.34  2057576.54
   19     14528    499.24  2044879.83
   20     15040    502.34  2057576.51
   21     15536    502.44  2057988.01
   22     16016    496.04  2031771.59
   23     16512    495.64  2030147.48
   24     17024    499.14  2044470.91
   25     17520    496.14  2032178.05

Reply:
MGR/dashboard component needs to update the UI to support the new burst seconds options.
https://tracker.ceph.com/issues/45641